lib/access-context.js

```diff
@@ -126,9 +131,29 @@ AccessContext.prototype.addPrincipal = function(principalType, principalId, prin
 AccessContext.prototype.getUserId = function() {
   for (var i = 0; i < this.principals.length; i++) {
     var p = this.principals[i];
+    if (p.type === Principal.APP || p.type === Principal.ROLE || p.type == Principal.SCOPE) {
+      continue;
+    }
+    // the principalType must either be 'USER'
     if (p.type === Principal.USER) {
       return p.id;
     }
+    // or permit to resolve a valid user model
+    var userModel = this.registry.findModel(p.type);
+    if (!userModel) continue;
+    if (userModel.modelName === builtinUserModel.name) {
```

Comment:
Just hardcode `userModel.modelName === 'User'` and make sure there is a test that would fail when this condition is not triggered (e.g. because the name of the built-in User model was changed).